test/libsolidity/SemanticTest.cpp

```diff
@@ -151,12 +162,99 @@ map<string, Builtin> SemanticTest::makeBuiltins() const
 			[this](FunctionCall const& _call) -> optional<bytes>
 			{
 				soltestAssert(_call.arguments.parameters.empty(), "No arguments expected.");
-				  return toBigEndian(u256(storageEmpty(m_contractAddress) ? 1 : 0));
-		 	}
+				return toBigEndian(u256(storageEmpty(m_contractAddress) ? 1 : 0));
+			}
+		},
+	};
+}
+
+vector<SideEffectHook> SemanticTest::makeSideEffectHooks() const
+{
+	return {
+		[](FunctionCall const& _call) -> vector<string>
+		{
+			if (_call.signature == "isoltest_side_effects_test")
+			{
+				vector<string> result;
+				for (auto const& argument: _call.arguments.parameters)
+					result.emplace_back(toHex(argument.rawBytes));
+				return result;
+			}
+			return {};
+		},
+		[this](FunctionCall const& _call) -> vector<string>
+		{
+			return eventSideEffectHook(_call);
 		}
 	};
 }
 
+vector<string> SemanticTest::eventSideEffectHook(FunctionCall const&) const
+{
+	static ABIType abiType(ABIType::Type::Hex);
+	vector<string> sideEffects;
+	vector<LogRecord> recordedLogs = ExecutionFramework::recordedLogs();
+	for (auto const& log: recordedLogs)
+	{
+		optional<EventInformation> event;
+		if (!log.topics.empty())
+			event = findEvent(log.topics[0]);
+		if (event.has_value())
+			soltestAssert(event.value().indexedTypes.size() == log.topics.size() - 1, "");
+
+		stringstream sideEffect;
+		sideEffect << "emit ";
+		if (event.has_value())
+			sideEffect << event.value().signature;
+		else
+			sideEffect << "<anonymous>";
+
+		if (m_contractAddress != log.creator)
+			sideEffect << " creator=0x" << log.creator;
+
+		vector<string> eventStrings;
+		for (auto& topic: log.topics)
```

Comment:
```suggestion
		for (h256 const& topic: log.topics)
```

There are also a few `auto` instances in `findEvent()` below that are just simple types so it would be nice to replace them too.